src/main/java/seedu/address/logic/commands/MarkPaidCommand.java

```diff
@@ -65,6 +66,30 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToMark = lastShownList.get(index.getZeroBased());
+        Set<MonthPaid> existingMonthsPaid = personToMark.getMonthsPaid();
+
+        // Define the pattern for valid YYYY-MM format
+        Pattern monthPattern = Pattern.compile("^\\d{4}-(0[1-9]|1[0-2])$");
+
+        // Check for invalid months paid format
+        for (MonthPaid monthPaid : monthsPaid) {
+            // Strip brackets before validation
+            String monthStr = monthPaid.toString().replaceAll("[\\[\\]]", "");
+            if (!monthPattern.matcher(monthStr).matches()) {
+                String invalidMonthsMessage = String.format("Invalid month format: %s. "
+                        + "Month must be in YYYY-MM format, where MM is 01-12.", monthStr);
+                throw new CommandException(invalidMonthsMessage);
+            }
+        }
+
+        //Check for duplicate months paid
+        for (MonthPaid monthPaid : monthsPaid) {
+            if (existingMonthsPaid.contains(monthPaid)) {
+                String duplicateMonthsMessage = String.format("Duplicate month paid: %s", monthPaid);
+                throw new CommandException(duplicateMonthsMessage);
+            }
+        }
```

Comment:
Don't need to check for duplicates as MonthsPaid is a Set of MonthPaid, and duplicates cannot be added to a Set